c10/core/TensorImpl.cpp

```diff
@@ -31,9 +31,19 @@ static void noop_decref_fn(const PyInterpreter*, PyObject*) {
   // no-op
 }
 
+static c10::intrusive_ptr<TensorImpl> noop_detach_fn(const PyInterpreter*, const TensorImpl*) {
+  return c10::intrusive_ptr<TensorImpl>();
```

Comment:
Not sure. Returning a nullptr here is OKish because `shallow_copy_and_detach` tests to see if a nullptr was returned, and if it was, does the "traditional" shallow copy and detach logic. So your code will work about as well as it possibly could given that the PyInterpreter is dead. It's true we could raise an error, maybe that is safer, but smoothing over the system here was pretty easy and predictable so I did it.

Reply:
Let's just raise an error; even if you successfully shallow copy and detach, if you try to run any op on it you are still going to die.